docs/tutorial/source-plugin-tutorial.md

```diff
@@ -0,0 +1,984 @@
+
+#### Adding dependencies
+
+You'll use several modules from npm to making fetching data with GraphQL easier. Install them with:
+
+```shell
+npm install apollo-cache-inmemory apollo-client apollo-link apollo-link-http apollo-link-ws apollo-utilities graphql graphql-tag node-fetch ws
+```
+
+_Note: The libraries used here are specifically chosen so that the source plugin can support GraphQL subscriptions. You can fetch data the same way you would in any other Node.js app or however you are most comfortable._
+
+Open your `package.json` file after installation and you'll see the packages have been added to a `dependencies` section at the end of the file.
+
+#### Configure an Apollo client to fetch data
+
+Import the handful of Apollo packages that you installed that help setup an Apollo client in your plugin:
+
+```javascript:title=source-plugin/gatsby-node.js
+// highlight-start
+const { ApolloClient } = require("apollo-client")
+const { InMemoryCache } = require("apollo-cache-inmemory")
+const { split } = require("apollo-link")
+const { HttpLink } = require("apollo-link-http")
+const { WebSocketLink } = require("apollo-link-ws")
+const { getMainDefinition } = require("apollo-utilities")
+const fetch = require("node-fetch")
+const gql = require("graphql-tag")
+const WebSocket = require("ws")
+// highlight-end
+
+const POST_NODE_TYPE = `Post`
+
+exports.sourceNodes = async ({
+// ...
+```
+
+Then you can copy this code that sets up the necessary pieces of the Apollo client and paste it after your imports:
+
+```javascript:title=source-plugin/gatsby-node.js
+// ... imports
+const WebSocket = require("ws")
+
+const POST_NODE_TYPE = `Post`
+
+// highlight-start
+const client = new ApolloClient({
+  link: split(
+    ({ query }) => {
+      const definition = getMainDefinition(query)
+      return (
+        definition.kind === "OperationDefinition" &&
+        definition.operation === "subscription"
+      )
+    },
+    new WebSocketLink({
+      uri: `ws://localhost:4000`, // or `ws://gatsby-source-plugin-api.glitch.me/`
+      options: {
+        reconnect: true,
+      },
+      webSocketImpl: WebSocket,
+    }),
+    new HttpLink({
+      uri: "http://localhost:4000", // or `https://gatsby-source-plugin-api.glitch.me/`
+      fetch,
+    })
+  ),
+  cache: new InMemoryCache(),
+})
+// highlight-end
+
+exports.sourceNodes = async ({
+// ...
+```
+
+You can read about each of the packages that are working together in [Apollo's docs](https://www.apollographql.com/docs/react/), but the end result is creating a `client` that you can use to call methods like `query` on to get data from the source it's configured to work with. In this case, that is `localhost:4000` where you should have the API running. If you can't configure the API to run locally, you can update the URLs for the client to use `gatsby-source-plugin-api.glitch.me` where a version of the API is deployed, instead of `localhost:4000`.
```

Comment:
```suggestion
You can read about each of the packages that are working together in [Apollo's docs](https://www.apollographql.com/docs/react/). The end result is creating a `client` that you can use to call methods like `query` to get data from the source it's configured to work with. In this case, that is `localhost:4000` where you should have the API running. If you can't configure the API to run locally, you can update the URLs for the client to use `gatsby-source-plugin-api.glitch.me` where a version of the API is deployed, instead of `localhost:4000`.
```